File: audio-streaming-client-java/src/main/java/com/baidu/acu/pie/model/AsrProduct.java
// Copyright (C) 2018 Baidu Inc. All rights reserved.

package com.baidu.acu.pie.model;

import lombok.Getter;

/**
 * AsrProduct
 *
 * @author Cynric Shu (cynricshu@gmail.com)
 */
public class AsrProduct {
    // 设置一些product兼容旧版本使用enum的情况
    public static AsrProduct CUSTOMER_SERVICE = new AsrProduct("客服模型", "1903", 8000);
    public static AsrProduct CUSTOMER_SERVICE_TOUR = new AsrProduct("客服模型：旅游领域", "1904", 8000);
    public static AsrProduct CUSTOMER_SERVICE_STOCK = new AsrProduct("客服模型", "1903", 8000);
    public static AsrProduct CUSTOMER_SERVICE_FINANCE = new AsrProduct("客服模型：股票领域", "1905", 8000);
    public static AsrProduct CUSTOMER_SERVICE_ENERGY = new AsrProduct("客服模型：能源领域", "1907", 8000);
    public static AsrProduct INPUT_METHOD = new AsrProduct("输入法模型", "888", 16000);
    public static AsrProduct FAR_FIELD = new AsrProduct("远场模型", "1888", 16000);
    public static AsrProduct FAR_FIELD_ROBOT = new AsrProduct("远场模型：机器人领域", "1889", 16000);
    public static AsrProduct SPEECH_SERVICE = new AsrProduct("演讲模型", "1912", 16000);

    @Getter
    private String name;

    @Getter
    private String code;

    @Getter
    private int sampleRate;

    public AsrProduct(String name, String code, int sampleRate) {
        this.name = name;
        this.code = code;
        this.sampleRate = sampleRate;
    }

    public AsrProduct(String code, int sampleRate) {
        this("自定义模型", code, sampleRate);
    }
}

File: audio-streaming-client-python/baidu_acu_asr/asr_product.py
# -*- coding: utf-8 -*-
from enum import Enum


class ProductMap(object):

    map = {
        "1903": [8000, "客服模型"],
        "1904": [8000, "客服模型：旅游领域"],
        "1905": [8000, "客服模型：股票领域"],
        "1906": [8000, "客服模型：金融领域"],
        "1907": [8000, "客服模型：能源领域"],
        "888": [16000, "输入法模型"],
        "1888": [16000, "远场模型"],
        "1889": [16000, "远场模型-机器人领域"],
        "1": [16000, "远场模型-法院用"]
    }

    @classmethod
    def get(cls, product_id):
        return cls.map[product_id][0]


class AsrProduct(Enum):
    CUSTOMER_SERVICE = ["客服模型", "1903", 8000]
    CUSTOMER_SERVICE_TOUR = ["客服模型：旅游领域", "1904", 8000]
    CUSTOMER_SERVICE_STOCK = ["客服模型：股票领域", "1905", 8000]
    CUSTOMER_SERVICE_FINANCE = ["客服模型：金融领域", "1906", 8000]
    CUSTOMER_SERVICE_ENERGY = ["客服模型：能源领域", "1907", 8000]
    INPUT_METHOD = ["输入法模型", "888", 16000]
    FAR_FIELD = ["远场模型", "1888", 16000]
    FAR_FIELD_ROBOT = ["远场模型：机器人领域", "1889", 16000]
    CHONGQING_FAYUAN = ["重庆高院", "1", 16000]